src/modules/commander/Commander.cpp

```diff
@@ -804,49 +804,57 @@ Commander::handle_command(vehicle_status_s *status_local, const vehicle_command_
 
 				bool cmd_arms = (static_cast<int>(cmd.param1 + 0.5f) == 1);
 
-				// Arm/disarm is enforced only when param2 is set to a magic number.
-				const bool enforce_in_air = (static_cast<int>(std::round(cmd.param2)) == 21196);
+				// Arm/disarm is enforced when param2 is set to a magic number.
+				const bool enforce = (static_cast<int>(std::round(cmd.param2)) == 21196);
 
-				if (!enforce_in_air && !land_detector.landed && !is_ground_rover(&status)) {
-					if (cmd_arms) {
-						mavlink_log_critical(&mavlink_log_pub, "Arming denied! Not landed");
-
-					} else {
-						mavlink_log_critical(&mavlink_log_pub, "Disarming denied! Not landed");
-					}
-
-					cmd_result = vehicle_command_s::VEHICLE_CMD_RESULT_DENIED;
-					break;
-				}
+				if (!enforce) {
```

Comment:
Could this be dangerous for folks who are already using the magic setting and don't expect that disarming is possible while the drone is flying?

Reply:
Disarming using the magic number was already possible. What I added now is arming.

Reply:
Ah never mind then